triage/script.js

```diff
@@ -0,0 +1,738 @@
+"use strict";
+
+var rightArrow = "\u25ba";
+var downArrow = "\u25bc";
+
+var builds = null;
+var clustered = null;		  // filtered clusters
+var clusteredAll = null;	  // all clusters
+var options = null;			  // user-provided in form or URL
+var lastClusterRendered = 0;  // for infinite scrolling
+
+const kGraphHeight = 200; 	  // keep synchronized with style.css:div.graph
+
+/**********************************************************************
+ *                        POLYFILLS                                   *
+ **********************************************************************/
+
+if (Object.entries === undefined) {
+	// Simple polyfill for Safari compatibility.
+	// Object.entries is an ES2017 feature.
+	Object.entries = function*(obj) {
+		for (let key of Object.keys(obj)) {
+			yield [key, obj[key]];
+		}
+	}
+}
+
+/**********************************************************************
+ *                                 DATASTRUCTURES                     *
+ **********************************************************************/
+
+// Return the minimum and maximum value of arr.
+// Doesn't cause stack overflows like Math.min(...arr).
+function minMaxArray(arr) {
+	var min = arr[0];
+	var max = arr[0];
+	for (var i = 1; i < arr.length; i++) {
+		if (arr[i] < min) min = arr[i];
+		if (arr[i] > max) max = arr[i];
+	}
+	return [min, max];
+}
+
+// Store information about individual builds.
+class Builds {
+	constructor(dict) {
+		this.jobs = dict.jobs;
+		this.jobPaths = dict.job_paths;
+		this.cols = dict.cols;
+		this.colStarted = this.cols.started;
+		this.colPr = this.cols.pr;
+		this.timespan = minMaxArray(this.cols.started);
```

Comment:
No, since the start and end times are more useful for graphing and other information display.